packages/server/modules/auth/defaultApps.js

```diff
@@ -57,7 +57,6 @@ async function updateDefaultApp(app, existingApp) {
   )
 
   if (scopeDiffA.length !== 0 || scopeDiffB.length !== 0) {
-    await revokeExistingAppCredentials({ appId: app.id })
```

Comment:
Okay - sorry for being late on this; we can do tomorrow am a meeting!  

This actually indeed prevents token revoking of default apps, but it leaves us in an invalid state: existing tokens will only have old scopes attached to them - when the app will try to use flex its new permissions, it won't actually be able to as its token does not hold the relevant scope (reverse is a problem too, token remains more powerful than it should be if we remove scopes). 

There's an extra tedious step in which we need to update all tokens that have been granted for this specific app to reflect the new scopes. 

`user_server_app_tokens` holds the info on what app what tokens, and `token_scopes` holds the token <> scopes relationship. Off the top of my head, I'd split this in two cases, addition and removal, and do some batch inserts/deletions if possible (or within one transaction) to prevent db chugging.